File: practice/hr-swap-two-num.js
function getNumOfSwapsToSort(array) {
  let sorted = [...array].sort((a, b) => a - b)
  // sorted
  let swaps = 0

  function swap() {
    let currentSwaps = 0
    array.forEach((el, i) => {
      if (array.length - 1 === i && currentSwaps !== 0) {
        swap()
      } else if (el !== sorted[i]) {
        let idx = sorted.indexOf(el)
        let temp = array[idx]
        array[idx] = el
        array[i] = temp
        swaps++
        currentSwaps++
      }
    })
  }

  swap()

  return swaps
}

const testArray2 = [3, 7, 6, 9, 1, 8, 10, 4, 2, 5]
const testArray = [7, 1, 3, 5, 2, 4, 6]
//

console.log(getNumOfSwapsToSort(testArray2))

File: random/hr-swap-two-num.py
def getNumOfSwapsToSort(array):
    sorted = array.copy()
    sorted.sort()

    swaps = 0

    def swap():
        currentSwaps = 0
        for i, el in enumerate(array):
            if ((len(array) - 1) == i) and currentSwaps != 0:
                swap()

            elif el != sorted[i]:
                idx = sorted.index(el)
                temp = array[idx]
                array[idx] = el
                array[i] = temp
                # UnboundLocalError: local variable 'swaps' referenced before assignment
                # if I use global swaps, then NameError: name 'swaps' is not defined
                nonlocal swaps
                swaps += 1
                currentSwaps += 1

    swap()

    return swaps


testArray2 = [3, 7, 6, 9, 1, 8, 10, 4, 2, 5]
testArray = [7, 1, 3, 5, 2, 4, 6]

print(getNumOfSwapsToSort(testArray))
# print('yo')
